foqus_lib/framework/optimizer/SM_Optimizer.py

```diff
@@ -0,0 +1,1191 @@
+            
+            initvals = []
+            decvars = [getattr(self.m,v) for v in dvar_names]
+            surroutvarlist = [str(v) for v in self.surrout_names_pyomo]
+            surroutvars = [getattr(self.m,v) for v in surroutvarlist]
+            
+            #Obtain the combination of all initialization points
+            for var in decvars:
+                initvals.append([var.value,var.lb,var.ub,(var.lb + var.ub)/2])
+            initvals_prod = list(product(*initvals))
+            
+            input_initvals = np.zeros(len(initvals_prod),dtype=object)
+            output_initvals = np.zeros(len(initvals_prod),dtype=object)
+            input_optim = np.zeros(len(initvals_prod),dtype=object)
+            output_optim = np.zeros(len(initvals_prod),dtype=object)
+            objvals = np.zeros(len(initvals_prod),dtype=object)
+            solver_status = np.zeros(len(initvals_prod),dtype=object)
+            solver_termination_condition = np.zeros(len(initvals_prod),dtype=object)
+            solver_time = np.zeros(len(initvals_prod),dtype=object)
+            
+            # Solve the optimization model for all initialization points & save solver details
+            for i1,initvallist in enumerate(initvals_prod):
+                i_initvals=[]
+                o_initvals=[]
+                i_optim=[]
+                o_optim=[]
+                for i,initv in enumerate(initvallist): 
+                    decvars[i].value = initv
+                    i_initvals.append(decvars[i].value)
+                for i,initout in enumerate(surroutvars):
+                    initout.value = -value(self.m.c[i+1].body - initout)
+                    o_initvals.append(initout.value)
+                input_initvals[i1] = i_initvals
+                output_initvals[i1] = o_initvals
+                r=optimizer.solve(self.m,**kwds)
+                for dvar in decvars:
+                    i_optim.append(dvar.value)
+                for sout in surroutvars:
+                    o_optim.append(sout.value)
+                if str(r.solver.termination_condition) == 'optimal':
+                    # input_initvals[i1] = i_initvals
+                    # output_initvals[i1] = o_initvals
+                    input_optim[i1] = i_optim
+                    output_optim[i1] = o_optim
+                    # if str(r.solver.termination_condition) == 'optimal':
+                    objvals[i1] = self.m.obj()
+                    solver_status[i1] = str(r.solver.status)
+                    solver_termination_condition[i1] = str(r.solver.termination_condition)
+                    solver_time[i1] = str(r.solver.time)
+                else:
+                    input_optim[i1] = 1e10
+                    output_optim[i1] = 1e10
+                    objvals[i1] = 1e10
+                    solver_status[i1] = 1e10
+                    solver_termination_condition[i1] = 1e10
+                    solver_time[i1] = 1e10
```

Comment:
Maybe see https://docs.python.org/3/library/math.html#math.inf depending on what you want to do here.  I assumed you just want to start optimal values out really big.  Not sure if inf is appropriate or not, so don't put to much stock in this.

Reply:
Thank you John. I will look into this and get back to you.

Reply:
I have replaced 1e10 with math.inf. It should work.